include/sys/fs/zfs.h

```diff
@@ -741,6 +766,11 @@ typedef enum zpool_errata {
 	ZPOOL_ERRATA_ZOL_2094_ASYNC_DESTROY,
 } zpool_errata_t;
 
+/* Include ARRAY_SIZE for userspace */
+#ifndef ARRAY_SIZE
+#define	ARRAY_SIZE(a) (sizeof (a) / sizeof (a[0]))
+#endif
```

Comment:
ARRAY_SIZE is a Linux kernel specific `#define`.  How about instead of using it directly we add ZFS_ARRAY_SIZE to `zfs_context.h` and define it as ARRAY_SIZE for the kernel, and then in the user space portion of this header define it as you've done here.